liquibase-core/src/main/java/liquibase/command/CommandArgumentDefinition.java

```diff
@@ -112,8 +114,12 @@ public String getDefaultValueDescription() {
      */
     public void validate(CommandScope commandScope) throws CommandValidationException {
         final DataType currentValue = commandScope.getArgumentValue(this);
+        // This is a list of the arguments which the init project command supports. The thinking here is that if the user
+        // forgets to supply one of these arguments, we're going to remind them about the init project command, which
+        // can help them figure out what they should be providing here.
+        final List<String> initProjectArguments = Arrays.asList(CommandArgumentNames.CHANGELOG_FILE, CommandArgumentNames.URL, CommandArgumentNames.USERNAME, CommandArgumentNames.PASSWORD);
```

Comment:
This seems like the wrong spot in the code to be putting this. The general CommandArgumentDefinition needs to be generic and not know anything about implementations. Adding logic for init project here is adding dependencies in the wrong direction.

If the init project doesn't have those as required, it should say that itself. Or if we are finding a use case that isn't currently handled by our CommandArgument metadata we should expand the metadata that can be attached to the arguments globally vs. hard-coding init into here.

Reply:
@nvoxland The acceptance criteria of the story which required this work says that we should be changing this message: "Error parsing command line: Invalid argument '--changelog-file': missing required argument", **for all commands** to "Error parsing command line: Invalid argument '--changelog-file': missing required argument. If you need to configure new liquibase project files and arguments, run the 'liquibase init project' command."

With that information, putting this logic elsewhere would require it be duplicated in many different places, wouldn't it? In other words, we want it here, because we want it to apply to every command which uses an argument which is also supported by `init project`.

Reply:
I figured it was something around that, but I wasn't sure how many of the arguments it should show that message on.

Having gotten rid of the extra parameter on the exception has helped, since this code is at least encapsulated for now. Ideally it maybe should be in the CLI code since it just talks about CLI functionality (even though this code is integration-agnostic). Or could be a separate CommandStep along side the init code that catches the exception and outputs the message. 

But not sure if either are worth the effort or not. The encapsulation makes it fine if we want.

Reply:
I moved this logic up a level, to the CLI code. Seems worth it to me.